src/dkg/round1.rs

```diff
@@ -135,32 +158,21 @@ pub fn import_secret_package(
 }
 
 #[must_use]
-fn input_checksum<I>(min_signers: u16, participants: &[I]) -> Checksum
+pub(super) fn input_checksum<'a, I>(min_signers: u16, participants: I) -> Checksum
 where
-    I: Borrow<Identity>,
+    I: IntoIterator<Item = &'a Identity>,
 {
-    // This function is only used in `PublicPackage::new()`, which in turn is only used in
-    // `round1()`. `round1()` already takes care of sorting and deduping the participants, hence we
-    // can assume that our input does not need any further processing. The following checks that
-    // it's indeed the case, but only for debug builds.
-    #[cfg(debug_assertions)]
-    {
-        let input_participants = participants.iter().map(Borrow::borrow).collect::<Vec<_>>();
-        let mut deduped_participants = input_participants.clone();
-        deduped_participants.sort_unstable();
-        deduped_participants.dedup();
-        debug_assert_eq!(
-            input_participants, deduped_participants,
-            "participants is expected to be sorted and to contain no duplicates"
-        );
-    }
```

Comment:
This optimization had to go away because now `input_checksum` is called in multiple places, and the precondition is not always true